lib/module.js

```diff
@@ -38,37 +41,41 @@ module.exports = function sentry(moduleOptions) {
     return
   }
 
-  // Initialize Sentry
-  Sentry.init({
-    dsn: options.dsn,
-    ...options.config
-  })
-  logger.success('Started logging errors to Sentry')
-
-  // Enable publishing of sourcemaps
-  this.extendBuild((config, { isClient, isDev }) => {
-    if (!options.publishRelease || isDev) return
-    if (isClient) config.devtool = '#sourcemap'
-
-    config.plugins.push(new WebpackPlugin(options.webpackConfig))
-    logger.info('Enabling uploading of release sourcemaps to Sentry')
-  })
-
   // Register the client plugin
   if (!options.disableClientSide) {
     this.addPlugin({
       src: path.resolve(__dirname, 'sentry.client.js'),
       fileName: 'sentry.client.js',
-      ssr: false,
       options: {
         dsn: options.dsn,
         ...options.config
       }
     })
   }
 
-  // Initialize the hooks
-  if (this.nuxt.hook) {
+  // Register the server plugin
+  if (!options.disableServerSide) {
+    // Initialize Sentry
+    Sentry.init({
+      dsn: options.dsn,
+      ...options.config,
+      integrations: [
+        new Integrations.Dedupe(),
+        new Integrations.ExtraErrorData(),
+        new Integrations.RewriteFrames(),
+        new Integrations.Transaction()
+      ]
+    })
+    logger.success('Started logging errors to Sentry')
+
+    this.addPlugin({
+      src: path.resolve(__dirname, 'sentry.server.js'),
```

Comment:
Will the name here, imply only including in server bundle?

Reply:
I suggest to specify `mode: 'server'` (Also this is not covering all 2.x support)

Reply:
@pi0 Yes, it should be processed by normalizePlugins as server only plugin, unless i miss something :)

Reply:
@pi0 hm, right . Is mode:server avaiable from 2.0 ? I think it was added later no?

Reply:
Yes both added from a minor version. Using explicit mode would be better idea.

Reply:
I changed to use explicit mode, but since mode available only from some minor it means we will only support it from there. So need change readme. Or maybe u have better ideas?

Reply:
I think mentioning in the readme/changelog is sufficient. BW support with alias makes module code complex and unmaintainable.